File: BukkitUtil-core/src/test/java/no/kh498/util/jackson/OfflinePlayerTest.java
package no.kh498.util.jackson;

import org.bukkit.Bukkit;
import org.bukkit.Location;
import org.bukkit.OfflinePlayer;
import org.bukkit.entity.Player;
import org.junit.Test;

import java.util.Map;
import java.util.UUID;

import static org.powermock.api.mockito.PowerMockito.when;

/**
 * @author Elg
 */
public class OfflinePlayerTest extends BukkitSerTestHelper {


    private static class OfflinePlayerTestImpl implements OfflinePlayer {

        UUID uuid;

        @Override
        public boolean isOnline() {
            return false;
        }

        @Override
        public String getName() {
            return null;
        }

        @Override
        public UUID getUniqueId() {
            return uuid;
        }

        @Override
        public boolean isBanned() {
            return false;
        }

        @Override
        public void setBanned(boolean banned) {

        }

        @Override
        public boolean isWhitelisted() {
            return false;
        }

        @Override
        public void setWhitelisted(boolean value) {

        }

        @Override
        public Player getPlayer() {
            return null;
        }

        @Override
        public long getFirstPlayed() {
            return 0;
        }

        @Override
        public long getLastPlayed() {
            return 0;
        }

        @Override
        public boolean hasPlayedBefore() {
            return false;
        }

        @Override
        public Location getBedSpawnLocation() {
            return null;
        }

        @Override
        public Map<String, Object> serialize() {
            return null;
        }

        @Override
        public boolean isOp() {
            return false;
        }

        @Override
        public void setOp(boolean value) {

        }
    }

    @Test
    public void serFromUUID() {

        UUID playerUUID = UUID.randomUUID();

        OfflinePlayerTestImpl player = new OfflinePlayerTestImpl();
        player.uuid = playerUUID;

        when(Bukkit.getOfflinePlayer(playerUUID)).thenReturn(player);

        testSerAll(player);
    }
}

File: BukkitUtil-core/src/test/java/no/kh498/util/jackson/OfflinePlayerTest.kt
package no.kh498.util.jackson

import com.fasterxml.jackson.core.JsonProcessingException
import org.bukkit.OfflinePlayer
import org.bukkit.craftbukkit.v1_x_Ry.JacksonMockServer
import org.junit.Assert
import org.junit.Test
import org.mockito.Mockito.`when`
import org.powermock.api.mockito.PowerMockito.mock
import java.util.*

/**
 * @author Elg
 */
class OfflinePlayerTest : BukkitSerTestHelper() {

    @Test
    @Throws(JsonProcessingException::class)
    fun serFromUUID() {
        val playerUUID = UUID.randomUUID()

        val mockPlayer = mock(OfflinePlayer::class.java)
        `when`(mockPlayer.uniqueId).thenReturn(playerUUID)
        JacksonMockServer.offlinePlayers += mockPlayer

        val player = mapper.readValue("\"$playerUUID\"", OfflinePlayer::class.java)
        Assert.assertSame(mockPlayer, player)
    }
}
